internal/internal_workflow_testsuite_test.go

```diff
@@ -1792,3 +1792,137 @@ func (s *WorkflowTestSuiteUnitTest) Test_ContextMisuse() {
 	s.Error(env.GetWorkflowError())
 	s.Contains(env.GetWorkflowError().Error(), "block on coroutine which is already blocked")
 }
+
+func (s *WorkflowTestSuiteUnitTest) Test_ActivityRetry() {
+
+	activityFn := func(ctx context.Context) (string, error) {
+		info := GetActivityInfo(ctx)
+		if info.Attempt < 2 {
+			return "", NewCustomError("bad-luck")
+		}
+		return "retry-done", nil
+	}
+
+	workflowFn := func(ctx Context) (string, error) {
+		ao := ActivityOptions{
+			ScheduleToStartTimeout: time.Minute,
+			StartToCloseTimeout:    time.Minute,
+			RetryPolicy: &RetryPolicy{
+				MaximumAttempts:          3,
+				InitialInterval:          time.Second,
+				MaximumInterval:          time.Second * 10,
+				BackoffCoefficient:       2,
+				NonRetriableErrorReasons: []string{"bad-error"},
+				ExpirationInterval:       time.Minute,
+			},
+		}
+		ctx = WithActivityOptions(ctx, ao)
+
+		var result string
+		err := ExecuteActivity(ctx, activityFn).Get(ctx, &result)
+		if err != nil {
+			return "", err
+		}
+		return result, nil
+	}
+
+	env := s.NewTestWorkflowEnvironment()
+	RegisterWorkflow(workflowFn)
+	RegisterActivity(activityFn)
+	env.ExecuteWorkflow(workflowFn)
+
+	s.True(env.IsWorkflowCompleted())
+	s.NoError(env.GetWorkflowError())
+	var result string
+	s.NoError(env.GetWorkflowResult(&result))
+	s.Equal("retry-done", result)
+}
+
+func (s *WorkflowTestSuiteUnitTest) Test_LocalActivityRetry() {
+
+	localActivityFn := func(ctx context.Context) (string, error) {
+		info := GetActivityInfo(ctx)
+		if info.Attempt < 2 {
+			return "", NewCustomError("bad-luck")
+		}
+		return "retry-done", nil
```

Comment:
May be return the non retryable error? Or have a test just for it?

Reply:
will update to cover non-retryable error case.